python-sdk/src/astro/databases/snowflake.py

```diff
@@ -622,13 +645,48 @@ def load_file_to_table_natively(
         )
         self.evaluate_results(rows)
 
+    def _get_copy_into_with_metadata_sql_statement(
+        self, file_path: str, target_table: BaseTable, stage: SnowflakeStage
+    ) -> str:
+        """Return the sql statement for copy into with metadata columns."""
+        if self.load_options is None or not self.load_options.metadata_columns:
+            raise ValueError("Error: Requires metadata columns to be set in load options")
+        table_name = target_table.name.upper()
+        sql_statement = (
+            "SELECT count(*) COLUMN_COUNT from INFORMATION_SCHEMA.columns where table_name=%(table_name)s"
+        )
+        try:
+            table_columns_count = self.hook.run(
+                sql_statement, parameters={"table_name": table_name}, handler=lambda cur: cur.fetchone()
+            )[0]
+        except AttributeError:  # pragma: no cover
+            # For apache-airflow-providers-snowflake <3.2.0.
```

Comment:
Any reason why it's less than 3.2.0? We should document this.

Reply:
It the same reason as is for other places in Snowflake in the same file. Also added a detailed note here too.